r2i/onnxrt/engine.cc

```diff
@@ -180,34 +204,72 @@ std::shared_ptr<r2i::IPrediction> Engine::Predict (std::shared_ptr<r2i::IFrame>
   return prediction;
 }
 
-size_t Engine::GetSessionInputCount(std::shared_ptr<Ort::Session> session) {
+size_t Engine::GetSessionInputCount(std::shared_ptr<Ort::Session> session,
+                                    RuntimeError &error) {
+  if (nullptr == session) {
+    error.Set (RuntimeError::Code:: NULL_PARAMETER,
+               "Received null session pointer");
+    return 0;
+  }
   return session->GetInputCount();
 }
 
-size_t Engine::GetSessionOutputCount(std::shared_ptr<Ort::Session> session) {
+size_t Engine::GetSessionOutputCount(std::shared_ptr<Ort::Session> session,
+                                     RuntimeError &error) {
+  if (nullptr == session) {
+    error.Set (RuntimeError::Code:: NULL_PARAMETER,
+               "Received null session pointer");
+    return 0;
+  }
   return session->GetOutputCount();
 }
 
 std::vector<int64_t> Engine::GetSessionInputNodeDims(
-  std::shared_ptr<Ort::Session> session, size_t index) {
-  return session->GetInputTypeInfo(
+  std::shared_ptr<Ort::Session> session, size_t index, RuntimeError &error) {
+  std::vector<int64_t> dims;
+  if (nullptr == session) {
+    error.Set (RuntimeError::Code:: NULL_PARAMETER,
+               "Received null session pointer");
+    return dims;
+  }
+  dims = session->GetInputTypeInfo(
            index).GetTensorTypeAndShapeInfo().GetShape();
+  return dims;
 }
 
 size_t Engine::GetSessionOutputSize(std::shared_ptr<Ort::Session> session,
-                                    size_t index) {
+                                    size_t index, RuntimeError &error) {
+  if (nullptr == session) {
+    error.Set (RuntimeError::Code:: NULL_PARAMETER,
+               "Received null session pointer");
+    return 0;
+  }
   return session->GetOutputTypeInfo(
            index).GetTensorTypeAndShapeInfo().GetElementCount();
 }
 
 const char *Engine::GetSessionInputName(std::shared_ptr<Ort::Session> session,
-                                        size_t index, OrtAllocator *allocator) {
-  return session->GetInputName(index, allocator);
+                                        size_t index, OrtAllocator *allocator, RuntimeError &error) {
+  const char *name = "";
+  if (nullptr == session) {
+    error.Set (RuntimeError::Code:: NULL_PARAMETER,
+               "Received null session pointer");
+    return name;
+  }
+  name = session->GetInputName(index, allocator);
```

Comment:
Are you sure this returns a const char? it's odd it receives an allocator but returns a const ptr.